src/tests/core/engine/processors/abstract_processor_test.py

```diff
@@ -0,0 +1,125 @@
+import time
+import unittest
+from abc import ABCMeta
+from datetime import datetime
+from unittest.mock import Mock, patch
+
+from src.core.engine.processors.abstract_processor import AbstractProcessor
+from src.tests.utils.fixtures.app_engine_processor_context_fixture import \
+    build_app_engine_processor_context
+from src.tests.utils.fixtures.app_processor_config_fixture import \
+    build_app_engine_config
+
+
+class ChildProcessor(AbstractProcessor, metaclass=ABCMeta):
+    def __init__(self, app_processor_config=None, app_context=None):
+        super().__init__(
+            app_processor_config=app_processor_config, app_context=app_context
+        )
+
+    def process(self, delta_time):
+        print("process execute")
+
+
+class TestAbstractProcessor(unittest.TestCase):
+    @patch("threading.Thread.__init__", return_value=None)
+    def test_thread_init_was_call(self, test_thread_init):
+
+        app_engine_config = build_app_engine_config(
+            id="test config", interval=10, on_start=None
+        )
+        app_engine_processor_context = build_app_engine_processor_context([])
+        test_class = ChildProcessor(app_engine_config, app_engine_processor_context)
+        test_thread_init.assert_called_once()
+
+    def test_app_engine_config(self):
+        app_engine_config = build_app_engine_config(id="test config", interval=10)
+        test_class = ChildProcessor(app_engine_config)
+        self.assertEqual(test_class.app_processor_config.id, app_engine_config.id)
+
+    def test_app_engine_processor_context(self):
+        app_engine_processor_context = build_app_engine_processor_context([])
+        test_class = ChildProcessor(None, app_engine_processor_context)
+        self.assertEqual(
+            test_class.app_context.processors, app_engine_processor_context.processors
+        )
+
+    def test_self_pause_is_false(self):
+        test_class = ChildProcessor()
+        self.assertEqual(test_class.pause, False)
+
+    def test_self_destroy_is_false(self):
+        test_class = ChildProcessor()
+        self.assertEqual(test_class.destroy, False)
+
+    def test_last_execution_time(self):
+        app_engine_config = build_app_engine_config(id="test config", interval=10)
+        test_class = ChildProcessor(app_engine_config)
+        test_class.start()
+        self.assertAlmostEqual(test_class.last_execution_time, datetime.now())
+
+    def test_process_method_was_call_with(self):
+        app_engine_config = build_app_engine_config(id="test config", interval=10)
+        test_class = ChildProcessor(app_engine_config)
+
+        with patch.object(ChildProcessor, "process") as mock:
```

Comment:
Could you follow same pattern as before?

Reply:
Done.